src/analysis/ml.py

```diff
@@ -45,7 +47,13 @@ def summarize_networks(file_paths: Iterable[Union[str, PathLike]]) -> pd.DataFra
                 if len(parts) > 0:  # in case of empty line in file
                     node1 = parts[0]
                     node2 = parts[1]
-                    edges.append(NODE_SEP.join(sorted([node1, node2])))  # assumes edges are undirected
+                    direction = str(parts[3]).strip()
+                    if direction == "U":
+                        edges.append(UNDIR_CONST.join([node1, node2]))
```

Comment:
Do we still need to sort nodes for undirected edges? Otherwise we may not match edges A-B and B-A in different graphs.